plugins/woocommerce-blocks/tests/e2e/tests/product-collection/register-product-collection-tester.block_theme.spec.ts

```diff
@@ -434,6 +453,7 @@ test.describe( 'Product Collection registration', () => {
 		} );
 
 		// Product picker should be shown in Editor
+		await page.reload();
```

Comment:
I think this is the only thing that's missing. Without it, the test is actually broken and TBH I'm not sure how it ever passed before 😅. Anyway, the additional checks above should be safe to remove! 🙌

Reply:
Yeah I am also confused how it passed before, which makes me think it skipped some checks in-between.
To me it kind felled like it skipped over the `await expect( editorProductPicker ).toBeHidden();` maybe it was satisfied when the block it-self was still in a loading state. That from there it jumped to the last one when it did show up.

I couldn't reproduce the errors otherwise locally and it was a bit hard to tell from the E2E video. I did waiver to keep the checks in, or remove the `await expect( editorProductPicker ).toBeHidden();` with the check I added to see if the product blocks rendered.